src/utils.c

```diff
@@ -33,3 +33,21 @@ void uint32_t_to_char_array(uint32_t const input, char *output) {
     }
     output[pos] = '\0';
 }
+
+void convert_to_hex_str(char* destination,
+                        uint8_t* source,
```

Comment:
You can make source `const uint8_t *`

Reply:
```
void convert_to_hex_str(char* destination,
                        size_t destination_size,
                        uint8_t* source,
                        size_t source_size);
```
would look a bit more standard but that's neatpicking, keep what you prefer

Reply:
applied both suggestions :+1: